### PR TITLE
#164798162 Run Code Climate Coverage

### PR DESCRIPTION
**Pull request by dorothyas**
#### What does this PR do?

- Runs codeclimate test coverage and maintainability after tests have run with Travis.

#### Description of Task to be completed?

-  intergrate the repo with codeclimate.
- Add mantainability and coverage badges to the README.md file.

#### How can this be tested??

- https://codeclimate.com/github/andela/ah-frontend-sparta

#### What are the relevant pivotal tracker stories?

[#164798162](https://www.pivotaltracker.com/n/projects/2320796/stories/164798162)

#### Screen shots

Screenshot 2019-05-02 at 23 05 17

**Badges**

Screenshot 2019-05-02 at 23 17 18

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -24,6 +24,10 @@ module.exports = {
         "jest/no-focused-tests": "error",
         "jest/no-identical-title": "error",
         "jest/prefer-to-have-length": "warn",
-        "jest/valid-expect": "error"
+        "jest/valid-expect": "error",
+        "react":"",
```
Comment (adesege) [inline .eslintrc.js]: Why are there two `rules` properties? Also, what does the `react` rule do there?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 .idea/
 node_modules/
 coverage/
+package-lock.json
```
Comment (adesege) [inline .gitignore]: Do not ignore package-lock.json

File: package.json [hunk 3]
```diff
@@ -53,6 +53,12 @@
   "jest": {
     "snapshotSerializers": [
       "enzyme-to-json/serializer"
-    ]
+    ],
+    "eslint-plugin-jsx-a11y": "^6.2.1",
```
Comment (adesege) [inline package.json]: These packages should not be listed here. They should either be in `dependecies` or `devDependencies`

**Review by adesege — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (dorothyas) [reply .gitignore]: okay I will undo this

File: .eslintrc.js (hunk 1 above)
Comment (adesege) [reply .eslintrc.js]: You still haven't answered my question here?

File: package.json (hunk 3 above)
Comment (adesege) [reply package.json]: You haven't fixed this as well

File: .eslintrc.js [hunk 4]
```diff
@@ -29,7 +29,33 @@ module.exports = {
...
+        "SharedArrayBuffer": "readonly"
+    },
+    "parserOptions": {
+        "ecmaFeatures": {
+            "jsx": true
+        },
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+    },
+    "plugins": [
```
Comment (adesege) [inline .eslintrc.js]: Reformat this file to use an indentation of two

**Review by adesege — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 5]
```diff
@@ -29,6 +29,31 @@ module.exports = {
...
+      rules: {
+        "react/jsx-filename-extension": [1, { extensions: [".js", ".jsx"] }],
+        "jest/no-disabled-tests": "warn",
+        "jest/no-focused-tests": "error",
+        "jest/no-identical-title": "error",
+        "jest/prefer-to-have-length": "warn",
+        "jest/valid-expect": "error",
+        react: ""
+      },
+      rules: {
```
Comment (adesege) [inline .eslintrc.js]: You removed the comments here regarding the multiple `rules` property name. Why is that?

File: package.json (hunk 3 above)
Comment (adesege) [reply package.json]: You haven't fixed this as well

**Review by adesege — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 5 above)
Comment (dorothyas) [reply .eslintrc.js]: @adesege am not really sure which comments you are talking about here.

**Review by adesege — APPROVED** [review]

**Pull request merged**